File: DataStructure_Alogorithm/Algorithm/Inflearn_basic_cpp/25_석차구하기/main.cpp
#include<stdio.h>
#include<algorithm>
using namespace std;

int main() {

	int i, j, n, a[200], b[200];
	scanf("%d ", &n);
	for(i=1; i<=n; i++) {
		scanf("%d", &a[i]);
		b[i]=1;
	}

	for(i=1; i<=n; i++) {
		for(j=1; j<=n; j++) {
			if(a[j]>a[i]) b[i]++;
		}
	}

	for(i=1; i<=n; i++) {
		printf("%d ", b[i]);
	}
	

	return 0;
}

File: DataStructure_Alogorithm/Algorithm/Inflearn_basic_cpp/25_석차구하기/main.go
package main

import (
	"fmt"
	"os"
)

func main() {
	file, _ := os.Open("input.txt")
	defer file.Close()
	old := os.Stdin
	defer func() { os.Stdin = old }()
	os.Stdin = file

	// 같은 점수 동일 석차
	var N int
	fmt.Scanf("%d", &N)
	arr := make([]int, N)

	for i := 0; i < N; i++ {
		fmt.Scanf("%d", &arr[i])
	}

	result := make([]int, N)
	for i := 0; i < N; i++ {
		result[i]++
		for j := 0; j < N; j++ {
			if i == j {
				continue
			}
			if arr[i] < arr[j] {
				result[i]++
			}
		}
	}
	fmt.Println(result)
}
